### PR TITLE
Make vertical mode UI dim when disabled

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Testing
See snapshot

Simulator Screen Recording -  snapshot tester  iPhone 12 mini 16 4 - 2024-08-01 at 11 39 48

## Changelog
Not shipped

### CODE REVIEW

**Comment by porter-stripe** [comment]
It looks strange to me that the entire UI is dimmed, but the processing button isn't either, especially when it is disable too. Is this intentional? Also the wallet (Apple Pay and Link) buttons don't dim.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/WalletHeaderView.swift [hunk 1]
```diff
@@ -182,3 +182,17 @@ extension PaymentSheetViewController {
...
 }
+
+// MARK: - EventHandler
+extension PaymentSheetViewController.WalletHeaderView: EventHandler {
+    func handleEvent(_ event: STPEvent) {
+        switch event {
+        case .shouldEnableUserInteraction:
+            alpha = 1
+        case .shouldDisableUserInteraction:
+            alpha = 0.8
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/WalletHeaderView.swift:193]: How'd we land on 0.6 for the rest of the UI for 0.8 here?

Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/WalletHeaderView.swift:193]: I see some 0.5's too

**Review by porter-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/WalletHeaderView.swift (hunk 1 above)
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/WalletHeaderView.swift:193]: I just eyeballed it. Will get Hannah to review next week.

**Pull request merged**